client/extensions/zoninator/state/locks/reducer.js

```diff
@@ -0,0 +1,67 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+import { combineReducers, createReducer, keyedReducer } from 'state/utils';
+import {
+	ZONINATOR_REQUEST_LOCK,
+	ZONINATOR_REQUEST_LOCK_ERROR,
+	ZONINATOR_UPDATE_LOCK,
+} from '../action-types';
+
+const isRequesting = createReducer(
+	{},
+	{
+		[ ZONINATOR_REQUEST_LOCK ]: () => true,
+		[ ZONINATOR_REQUEST_LOCK_ERROR ]: () => false,
+		[ ZONINATOR_UPDATE_LOCK ]: () => false,
+	}
+);
+
+export const requesting = keyedReducer( 'siteId', keyedReducer( 'zoneId', isRequesting ) );
+
+const isBlocked = createReducer(
+	{},
+	{
+		[ ZONINATOR_UPDATE_LOCK ]: () => false,
+		[ ZONINATOR_REQUEST_LOCK_ERROR ]: ( state, { blocked } ) => blocked,
+	}
+);
+
+export const blocked = keyedReducer( 'siteId', keyedReducer( 'zoneId', isBlocked ) );
+
+const lockInit = createReducer(
+	{},
```

Comment:
Similar to https://github.com/Automattic/wp-calypso/pull/21110/files#r160455307, I'd avoid 'empty object' as an initial value for a reducer that's supposed to hold a flat data type. Since we're dealing with an integer here, would `0` work?